tutorials/3d/csg_tools.rst

```diff
@@ -91,15 +91,26 @@ The :ref:`CSGPolygon3D <class_CSGPolygon3D>` node extrude along a Polygon drawn
 Custom meshes
 ~~~~~~~~~~~~~
 
-Any mesh can be used for :ref:`CSGMesh3D <class_CSGMesh3D>`; the mesh can be
-modeled in other software and imported into Godot. Multiple materials are
-supported. There are some restrictions for geometry:
+Custom meshes can be used for :ref:`CSGMesh3D <class_CSGMesh3D>` as long as the
+mesh is *manifold*. The mesh can be modeled in other software and imported into
+Godot. Multiple materials are supported. A manifold mesh has some restrictions
+for geometry:
 
 - it must be closed,
 - it must not self-intersect,
 - it must not contain internal faces,
 - every edge must connect to only two other faces.
```

Comment:
The triangle vertices must appear clockwise when viewed from outside the Godot Engine manifold mesh.

It can be split into two layperson explanations.

1. The Godot Engine triangle mesh winding order should be clockwise.
2. All the windings should be clockwise. There is no mixing between clockwise and counterclockwise.

Reply:
@elalish Do you know if we need to explain something about normals? I didn't see that in your explanation.

Reply:
Technically the three of these except for self-intersection are more precisely defined by the manifold definition below. Self-intersection should be avoided, but Manifold is robust to it at least, and it's difficult to test in practice. 

As for winding direction - the manifoldness definition below already restricts us to an oriented 2-manifold, so there can't be any mixing of CW and CCW. Like self-intersection, CCW can be hard to test reliably. And manifold will handle inverted meshes just fine - it's simply that CW triangles will lead to negative volume. I would trust a positive volume check over checking CCW. Likewise, normals are not very trustworthy.

Reply:
Clarification for others:

![introduction_to_3d_coordinate_systems](https://github.com/user-attachments/assets/44d2942c-cf4b-4c28-9faf-befd2dfd47f4)

Godot Engine has a different handedness than `elalish/manifold`. So, the winding order is swapped.

Reply:
Hmm, Manifold is in the lower-right - Z-up and right-handed, so it looks like Godot has the same handedness? Or does Godot just use CW triangle winding as positive?

Reply:
https://docs.godotengine.org/en/latest/classes/class_arraymesh.html#arraymesh

> Note: Godot uses clockwise [winding order](https://learnopengl.com/Advanced-OpenGL/Face-culling) for front faces of triangle primitive modes.